mindmeld/models/taggers/pytorch_crf.py

```diff
@@ -273,66 +297,68 @@ def compute_marginal_probabilities(self, inputs, mask):
         prob = alpha + beta - z.view(1, -1, 1)
         return torch.exp(prob).transpose(0, 1)
 
-    def set_params(self, **params):
-        self.feat_type = params.get('feat_type', DEFAULT_PYTORCH_CRF_ER_CONFIG['feat_type']).lower()
-        self.feat_num = params.get('feat_num', DEFAULT_PYTORCH_CRF_ER_CONFIG['feat_num'])
-        self.stratify = params.get('stratify_train_val_split',
-                                   DEFAULT_PYTORCH_CRF_ER_CONFIG['stratify_train_val_split'])
-        self.drop_input = params.get('drop_input', DEFAULT_PYTORCH_CRF_ER_CONFIG['drop_input'])
-        self.train_batch_size = params.get('train_batch_size', DEFAULT_PYTORCH_CRF_ER_CONFIG['train_batch_size'])
-        self.patience = params.get('patience', DEFAULT_PYTORCH_CRF_ER_CONFIG['patience'])
-        self.epochs = params.get('epochs', DEFAULT_PYTORCH_CRF_ER_CONFIG['epochs'])
-        self.train_val_split = params.get('train_val_split', DEFAULT_PYTORCH_CRF_ER_CONFIG['train_val_split'])
-        self.optimizer_type = params.get('optimizer_type', DEFAULT_PYTORCH_CRF_ER_CONFIG['optimizer_type']).lower()
-        self.random_state = params.get('random_state', randint(1, 10000001))
+    def set_params(self, feat_type="hash", feat_num=50000, stratify_train_val_split=True, drop_input=0.2, batch_size=8,
+                   patience=3, number_of_epochs=100, dev_split_ratio=0.2, optimizer="sgd",
+                   random_state=randint(1, 10000001)):
```

Comment:
Here you'll probably want to set the default of random_state to `None` and then do something like:

```python
random_state = random_state or randint(1, 10000001)
```

As written the randint function will get called when the _class_ is created and then every function call would use the same random value, which I don't think is what you'd want.

Reply:
Fixed